### PR TITLE
CNV-38375: add hideLabelFilter and hideNameLabelFilters

### PR DESCRIPTION
**Pull request by upalatucci**
## 📝 Description

add   `hideLabelFilter?: boolean; hideNameLabelFilters?: boolean;` props.
Considering those props, render dynamically the selectoption in the select search type.
Considering those props, decide which option has to be selected at the beginning:

decision of the initial option BEFORE: 
if the user has already searched something, choose the searched type otherwise select NAME option

decision of the initial option WITH THIS PR:
 
if the user has already searched something, choose the searched type otherwise select `NAME` option ONLY if `NAME` is not hidden. If `NAME` option is hidden, choose a different search option

## 🎥 Demo

*Without props*
image

*With hideLabelFilter*
image

*With hideNameLabelFilters*

image

There is no select as we have only one option available so the user cannot change the search type here.

*With hideNameLabelFilters and no custom search*

image

**Fix also the category chip name**
Before was just `ip` now `IP Address` 
image

### CODE REVIEW

**Comment by metalice** [comment]
please uncomment the props that ive hidden in the router pr and add to this pr, check that its working

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 1]
```diff
@@ -135,107 +172,96 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
+                  <ToolbarFilter
+                    deleteChip={() => {
+                      applyTextFilters(filter.type);
+                      searchType === filter.type && setSearchInputText('');
+                    }}
+                    categoryName={filter.filterGroupName}
+                    chips={textFilters[filter.type] ? [textFilters[filter.type]] : []}
+                    key={filter.type}
+                  >
+                    <></>
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:198]: need that?

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 2]
```diff
@@ -135,107 +172,96 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
-                applyTextFilters(STATIC_SEARCH_FILTERS.labels);
-              }}
-              categoryName={STATIC_SEARCH_FILTERS_LABELS.labels}
-              chips={textFilters.labels ?? []}
-            >
-              {searchFilters.map((filter) => (
+          {showSearchFilters && (
+            <ToolbarItem className="co-filter-search--full-width">
+              <ToolbarFilter
+                deleteChip={(f, chip: string) => {
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:178]: f?

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 3]
```diff
@@ -117,6 +150,10 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
 
   const selectedSearchFilter = searchFilters?.find((f) => f.type === searchType);
 
+  const showSearchFilters = Object.keys(filterDropdownItems).length !== 0;
+
+  const showSearchFiltersDropdown = Object.keys(filterDropdownItems).length > 1;
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:155]: feels strange.. 1 check if its 1 or more the other check more then one? can u make it one? or use showSearchFiltersDropdown = showSearchFilter > 1

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 4]
```diff
@@ -83,12 +90,38 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
+    if (searchType === STATIC_SEARCH_FILTERS.name)
+      return nameFilterPlaceholder || STATIC_SEARCH_FILTERS_PLACEHOLDERS.name;
+
+    return (
+      STATIC_SEARCH_FILTERS_PLACEHOLDERS[searchType] ||
+      t('Search by {{filterName}}...', {
+        filterName: selectedSearchFilter?.filterGroupName,
+      })
+    );
+  };
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:103]: can this be extracted from this file as util?

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 5]
```diff
@@ -83,12 +90,38 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
+    ),
+  };
+
+  if (!hideLabelFilter && !hideNameLabelFilters) {
+    filterDropdownItems.labels = t('Label');
+  }
+
+  if (!hideNameLabelFilters) {
+    filterDropdownItems.name = t('Name');
+  }
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:121]: why not adding it to filterDropdownItems ? it as a reduce funtion

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 6]
```diff
@@ -135,107 +172,96 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
-              }}
-              categoryName={STATIC_SEARCH_FILTERS_LABELS.labels}
-              chips={textFilters.labels ?? []}
-            >
-              {searchFilters.map((filter) => (
+          {showSearchFilters && (
+            <ToolbarItem className="co-filter-search--full-width">
+              <ToolbarFilter
+                deleteChip={(f, chip: string) => {
+                  const newLabels = textFilters.labels.filter((label) => label !== chip);
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:179]: please add .? unless it gurantee

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 7]
```diff
@@ -135,107 +172,96 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
+                            {t('Filter')}
+                          </span>
+                        }
+                        isOpen={isDropdownOpen}
+                        onSelect={onSelect}
+                        onToggle={setIsDropdownOpen}
+                        selections={selectedSearchFilter?.filterGroupName || searchType}
+                        variant={SelectVariant.single}
+                      >
+                        {Object.keys(filterDropdownItems).map((key) => (
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:225]: its the 3rd time u are doing object keys on this object, maybe save it?

File: src/utils/components/ListPageFilter/ListPageFilter.tsx [hunk 8]
```diff
@@ -135,107 +172,96 @@ const ListPageFilter: FC<ListPageFilterProps> = ({
...
+                          <SelectOption key={key} value={key}>
+                            {filterDropdownItems[key]}
```
Comment (metalice) [inline src/utils/components/ListPageFilter/ListPageFilter.tsx:227]: maybe .?

File: src/utils/components/ListPageFilter/ListPageFilter.tsx (hunk 1 above)
Comment (upalatucci) [reply src/utils/components/ListPageFilter/ListPageFilter.tsx:200]: yes. `ToolbarFilter` requires a children

File: src/utils/components/ListPageFilter/ListPageFilter.tsx (hunk 5 above)
Comment (upalatucci) [reply src/utils/components/ListPageFilter/ListPageFilter.tsx:121]: name and label are not part of `searchFilters`. Name and label are added by default no matter the `searchFilters` state. To remove them you have to use `hideNameLabelFilters` and `hideLabelFilter`

**Pull request merged**